### DISCUSSION TITLE
Proposal: An updated hierarchical facets behaviour

### DISCUSSION THREAD

**Discussion by vvo** [algolia/instantsearch#5732] [Ideas]
# Disjunctive facet behaviour

When using a **menu** in React InstantSearch (based on a **disjunctive facet**), we get this behaviour:

![2018-01-25 14 10 40](https://user-images.githubusercontent.com/123822/35390971-8105b866-01dd-11e8-8f79-845fdb6f6331.gif)

Steps to reproduce:
- Load the [Menu story](https://community.algolia.com/react-instantsearch/storybook/?selectedKind=Menu&selectedStory=default&full=0&addons=1&stories=1&panelRight=1&addonPanel=storybooks%2Fstorybook-addon-knobs)
- Click on Lighting
- Type/search "boholmen"

Expected result: even if lightning has 0 matching items, other categories are here to give context about results.

Actual result: exactly the expected result

# Hierarchical facet behaviour

When using a **hierarchical menu** in React InstantSearch (based on a **hierarchical facet**), we get this behaviour:

![2018-01-25 14 41 37](https://user-images.githubusercontent.com/123822/35391093-ebcb53c2-01dd-11e8-80f2-7dd70862bb29.gif)

Steps to reproduce:
- Load the [HierarchicalMenu story](https://community.algolia.com/react-instantsearch/storybook/?selectedKind=HierarchicalMenu&selectedStory=default&full=0&addons=1&stories=1&panelRight=1&addonPanel=storybooks%2Fstorybook-addon-knobs)
- Click on Lighting
- Type/search "boholmen"

Expected result: even if lightning has 0 matching items, other categories are here to give context about results.

Actual result: the widget is disappearing because we are not searching in other facets

This behaviour impacts all InstantSearch flavours.

Should we try to fix that? I believe yes if doable. This would be a "breaking change" but since the new behaviour would be better in term of user experience, that should be considered as minor.

**Comment by sreidaf** [comment]
Hi - any update on this ticket, or workaround ?
